### PR TITLE
Fix/api change compatibility

### PR DESCRIPTION
**Pull request by jackraymond**
Examples and tests are updated to fix breakages caused by changes to the get_properties return format.

Defaults are modified to accommodate the change from Advantage2_research1 to Advantage2_research2.

This pull request contains a subset of changes in https://github.com/dwavesystems/dwave-experimental/pull/51.

### CODE REVIEW

File: examples/mca_shim_AO_FB.py
Comment (thisac) [inline examples/mca_shim_AO_FB.py]: Need to update this as well.

**Review by thisac — APPROVED** [review]
LGTM. I noticed that there's some coverage missing for the mca tests, but this might be better addressed separately (specifically exceptions aren't being tested + some edge cases).

File: releasenotes/notes/get_properties_and_research2_fixes-0275ae3d8bb6dd8c.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+fixes:
+  - |
+    Update examples, tests and utils to accommodate modified get_properties
+    return format.
```
Comment (randomir) [inline releasenotes/notes/get_properties_and_research2_fixes-0275ae3d8bb6dd8c.yaml:5]: Can you also please update the `get_properties()` (return type, docs) to reflect the new format returned by the server?

**Review by randomir — COMMENTED** [review]
I'm OK with merging this, but we should improve the return format of `mca.get_properties()` in a follow-up PR.

Returning a list of unrelated items (as SAPI does now) is really not ideal.

Opened #61.
